skopt/acquisition.py

```diff
@@ -264,3 +267,119 @@ def gaussian_ei(X, model, y_opt=0.0, xi=0.01, return_grad=False):
+
+    1. It approximates the search space by a set of representer points sampled from
+       a suitable measure such as the "probability of improvement" or
+       "expected improvement".
+
+    2. p_min at a representer point X_i is calculated by the Integral
+       p(f(x)) \prod_{j=1}^N_rep \Theta(X_j - X_i) \\
+       Where P(f(x)) is the GP posterior and \Theta is the heaviside step function.
+       This can be used to calculate the base entropy.
+
+    3. Then for each candidate point, the change in the mean and the covariance
+       of each of these representer points can be computed. This in turn
+       can be used to compute the change in p_min for each of these representer
+       points.
+
+    Parameters
+    ----------
+    * `X` [array-like, shape=(n_samples, n_features)]:
+        Values where the acquisition function should be computed.
+
+    * `model` [sklearn estimator that implements predict with ``return_std``]:
+        The fit estimator that approximates the function through the
+        method ``predict``.
+        It should have a ``return_std`` parameter that returns the standard
+        deviation.
+
+    * `space` [Space instance]:
+        This is required to sample representer points.
+
+    * `n_rep_points` [int, default 50]:
+        Number of represter points.
+
+    * `n_trial_points` [int, default 200]:
+        Number of trial points sampled uniformly for each representer point
+        candidate.
+
+    * `n_samples` [int, default 500]:
+        Number of samples to sample from the GP posterior to approximate p_min
+
+    * `random_state` [int, RandomState instance, or None (default)]:
+        For reproducible results.
+    """
+    rng = np.random.RandomState(random_state)
+    n_features = X.shape[1]
+
+    # Step 1: Sample `n_representer_points` from the given nonmeasure.
+    # XXX: Use MCMC sampling to sample from the quadrature measure.
+    n_total = n_rep_points*n_trial_points
+    rep_cands = space.transform(space.rvs(n_samples=n_total, random_state=rng))
+    rep_points = np.zeros((n_rep_points, n_features))
+
+    if measure == "LCB":
+        quad_func = gaussian_lcb
+
+    elif measure in ["EI", "PI"]:
+        if measure == "EI":
+            quad_func = lambda X, model: -gaussian_ei(X, model)
+        else:
+            quad_func = lambda X, model: -gaussian_pi(X, model)
+
+    for i, startptr in enumerate(range(0, X.shape[0], n_total)):
+        rep_curr = rep_cands[startptr: startptr+n_trial_points]
+        argmin = np.argmin(quad_func(rep_curr, model))
+        rep_points[i] = rep_curr[argmin]
+
+    # # Step 2: Calculate p_min for each of these represnter points using
+    # # Monte-Carlo approximation.
+    # sampled_f = model.sample_y(
+    #     X_candidates, n_samples=n_samples, random_state=rng)
+    # p_min = (np.bincount(np.argmax(sampled_f, axis=0)) + 1) / (n_candidates + float(n_samples))
+    # base_entropy = entropy(p_min)
```

Comment:
It doesn't affect the maximizer of the information gain since - as you say - it is the same for all candidate points. However, if you want to interpret the value of the acquisition function as information gain, you need this "base" entropy. It is also relatively cheap as it has effectively only the cost of evaluating one additional query point. But you don't need it strictly. But then the returned quantity should not be named "inf_gain".